### PR TITLE
Reserve volume aggregation

### PR DESCRIPTION
**Pull request by cyberliem**

### CODE REVIEW

File: tradelogs/http/asset_volume_handler.go [hunk 1]
```diff
@@ -60,6 +60,13 @@ func (sv *Server) getAssetVolume(c *gin.Context) {
 }
 
 func timeValidation(fromTime, toTime *uint64, c *gin.Context, logger *zap.SugaredLogger) bool {
+	if *fromTime == 0 && *toTime == 0 {
```
Comment (favadi) [inline tradelogs/http/asset_volume_handler.go]: what is the reasoning behind this changes? Right be low we set default value when fromTime, toTime is not set, so why should we returns bad request here?

File: tradelogs/http/asset_volume_handler.go [hunk 2]
```diff
@@ -73,3 +80,16 @@ func timeValidation(fromTime, toTime *uint64, c *gin.Context, logger *zap.Sugare
 	}
 	return true
 }
+
+func (sv *Server) lookupToken(ID string) (core.Token, error) {
```
Comment (favadi) [inline tradelogs/http/asset_volume_handler.go]: there is `core.LookupToken` already.

File: tradelogs/http/reserve_volume_handler.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	}
+	if !timeValidation(&query.From, &query.To, c, logger) {
+		logger.Info("time validation returned invalid")
+		return
+	}
+	token, err := sv.lookupToken(query.Asset)
+	if err != nil {
+		c.JSON(
+			http.StatusInternalServerError,
+			gin.H{"error": err.Error()},
```
Comment (favadi) [inline tradelogs/http/reserve_volume_handler.go]: please rebase the PR on develop and use httputil.ResponseFailure.

Comment (favadi) [reply tradelogs/http/reserve_volume_handler.go]: same comments for all failure paths in this handler.

File: tradelogs/storage/cq/volume_cqs.go [hunk 4]
```diff
@@ -1,46 +1,55 @@
...
 import (
-	"github.com/KyberNetwork/reserve-stats/lib/cq"
+	"bytes"
+	"text/template"
+
+	libcq "github.com/KyberNetwork/reserve-stats/lib/cq"
+)
+
+const (
+	rsvVolTemplate    = `SELECT SUM({{.AmountType}}) AS token_volume, SUM(eth_amount) AS eth_volume, SUM(usd_amount) AS usd_volume INTO {{.MeasurementName}} FROM (SELECT {{.AmountType}}, eth_amount, eth_amount*eth_usd_rate AS usd_amount FROM trades WHERE ((src_addr!='0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE' AND dst_addr!='0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2') OR (src_addr!='0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2' AND dst_addr!='0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE')) AND {{.RsvAddressType}}!='') GROUP BY {{.AddressType}},{{.RsvAddressType}}`
```
Comment (favadi) [inline tradelogs/storage/cq/volume_cqs.go]: need comment about what is 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2. Also, please break long lines.

File: tradelogs/storage/cq/volume_cqs.go [hunk 5]
```diff
@@ -64,3 +73,95 @@ func CreateAssetVolumeCqs(dbName string) ([]*cq.ContinuousQuery, error) {
 	result = append(result, assetVolSrcDayCqs)
 	return result, nil
 }
+
+// RsvFieldsType declare the set of names requires to completed a reserveVolume Cqs
+type RsvFieldsType struct {
+	AmountType     string
+	RsvAddressType string
+	AddressType    string
```
Comment (favadi) [inline tradelogs/storage/cq/volume_cqs.go]: please add comment for all the fields of this struct as it is not very clear what to pass to them looking at the names.

File: tradelogs/storage/testdata/export.dat [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (favadi) [inline tradelogs/storage/testdata/export.dat]: we shouldn't change this file here, right?

File: tradelogs/storage/volume_influxdb.go [hunk 7]
```diff
@@ -21,13 +21,46 @@ const (
...
+	}
 )
 
+// GetReserveVolume returns the volume of a specific asset(token) from a reserve
+// between a period and with desired frequency
+func (is *InfluxStorage) GetReserveVolume(rsvAddr ethereum.Address, token core.Token, fromTime, toTime uint64, frequency string) (map[uint64]*common.VolumeStats, error) {
+	var (
+		rsvAddrHex   = rsvAddr.Hex()
+		tokenAddrHex = ethereum.HexToAddress(token.Address).Hex()
+		logger       = is.sugar.With("reserve Address", rsvAddr.Hex(), "token Address", token.Address, "from", fromTime, "to", toTime)
```
Comment (favadi) [inline tradelogs/storage/volume_influxdb.go]: please add function "func":  "tradelogs/storage/InfluxStorage.GetReserveVolume"

File: tradelogs/storage/volume_influxdb.go [hunk 8]
```diff
@@ -21,13 +21,46 @@ const (
...
+	mName, ok := rsvMeasurementName[strings.ToLower(frequency)]
+	if !ok {
+		return nil, fmt.Errorf("frequency %s is not supported", frequency)
+	}
+
+	addrFilter := fmt.Sprintf("((dst_addr='%s' OR src_addr='%s') AND (dst_rsv_addr='%s' OR src_rsv_addr='%s'))", tokenAddrHex, tokenAddrHex, rsvAddrHex, rsvAddrHex)
+	timeFilter := fmt.Sprintf("(time >=%d%s AND time <= %d%s)", fromTime, timePrecision, toTime, timePrecision)
+	cmd := fmt.Sprintf("SELECT SUM(token_volume) as %s, SUM(eth_volume) as %s, SUM(usd_volume) as %s FROM %s WHERE %s AND %s GROUP BY time(1%s) FILL(0)", tokenVolumeField, ethVolumeField, fiatVolumeField, mName, timeFilter, addrFilter, frequency)
+
+	logger.Debugf("the query is %s", cmd)
```
Comment (favadi) [inline tradelogs/storage/volume_influxdb.go]: please use Debugw like `logger.Debugw("query rendered", "cmd", cmd)`.

File: tradelogs/storage/cq/volume_cqs.go [hunk 9]
```diff
@@ -64,3 +73,95 @@ func CreateAssetVolumeCqs(dbName string) ([]*cq.ContinuousQuery, error) {
...
+			return nil, err
+		}
+		hourCQ, err := libcq.NewContinuousQuery(
+			name+"_hour",
+			dbName,
+			hourResampleInterval,
+			hourResampleFor,
+			query,
+			"1h",
+			[]string{},
```
Comment (favadi) [inline tradelogs/storage/cq/volume_cqs.go]: is it possible to pass nil instead of `[]string{}`?

File: tradelogs/storage/cq/volume_cqs.go [hunk 10]
```diff
@@ -64,3 +73,95 @@ func CreateAssetVolumeCqs(dbName string) ([]*cq.ContinuousQuery, error) {
...
+			return nil, err
+		}
+		dayCQ, err := libcq.NewContinuousQuery(
+			name+"_day",
+			dbName,
+			dayResampleInterval,
+			dayResampleFor,
+			query,
+			"1d",
+			[]string{},
```
Comment (favadi) [inline tradelogs/storage/cq/volume_cqs.go]: is it possible to pass nil instead of `[]string{}`?

File: tradelogs/storage/testdata/export.dat (hunk 6 above)
Comment (cyberliem) [reply tradelogs/storage/testdata/export.dat]: We must, getReserveVolume is a new function required to return the Reserve Volume from trade_logs influx.

**Pull request merged**